libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightAlertConfig/AlertComparisonPeriodSelect.tsx

```diff
@@ -0,0 +1,138 @@
+// (C) 2022-2024 GoodData Corporation
+import React, { useMemo } from "react";
+import { IAutomationMetadataObject } from "@gooddata/sdk-model";
+import { Button, Dropdown, List, OverlayPositionType, SingleSelectListItem } from "@gooddata/sdk-ui-kit";
+import { defineMessage, FormattedMessage, useIntl } from "react-intl";
+import cx from "classnames";
+
+import { AlertMetric, AlertMetricComparator, AlertMetricComparatorType } from "../../types.js";
+
+import { isChangeOrDifferenceOperator } from "./utils/guards.js";
+import { DROPDOWN_ITEM_HEIGHT } from "./constants.js";
+
+export interface IAlertComparisonPeriodSelectProps {
+    alert: IAutomationMetadataObject | undefined;
+    measure: AlertMetric | undefined;
+    overlayPositionType?: OverlayPositionType;
+    selectedComparison?: AlertMetricComparatorType;
+    onComparisonChange: (comparison: AlertMetricComparatorType) => void;
+    canManageComparison: boolean;
+}
+
+const comparisons = [
+    {
+        title: defineMessage({ id: "insightAlert.config.compare_with_sp" }),
+        type: AlertMetricComparatorType.SamePeriodPreviousYear,
+    },
+    {
+        title: defineMessage({ id: "insightAlert.config.compare_with_pp" }),
```

Comment:
Seems as same message used in FormattedMessage at bottom of the file, maybe we should deduplicate them?